frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/SyncSelectedToParentByMergeAction.java

```diff
@@ -1,16 +1,50 @@
 package com.virtuslab.gitmachete.frontend.actions.contextmenu;
 
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.ide.util.PropertiesComponent;
 import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.ui.Messages;
 import io.vavr.control.Option;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
 import com.virtuslab.gitmachete.frontend.actions.base.BaseSyncToParentByMergeAction;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.WarnAboutSyncToParentByMergeDialog;
 import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
 
 public class SyncSelectedToParentByMergeAction extends BaseSyncToParentByMergeAction
     implements
       IExpectsKeySelectedBranchName {
+
+  public static final String MERGE_INFO_SHOWN = "git-machete.merge.warning.shown";
```

Comment:
Note that UI tests now time out, and rightfully so: they're stuck on the dialog.

See how it was resolved in UI tests for the Reset dialog (I don't remember now, either the dialog is clicked automatically, or the corresponding `...shown` property is set programatically)

Reply:
Also... the name for the key (same for the Reset dialog's key) is a misnomer &mdash; the dialog could've been shown many times, and the key can still be `false`. It's not about whether it was **shown**, it's about whether it **should not be shown again** 🤔